js/client-battle.js

```diff
@@ -1083,6 +1112,238 @@
+				}
+				basePower = 60 + 20 * multiply;
+			}
+			if (move.id === 'smellingsalts') {
+				if (target.status === 'par') {
+					basePower *= 2;
+					basePowerComment = ' (Boosted by status)';
+				}
+			}
+			if (move.id === 'storedpower') {
+				var boosts = Object.keys(pokemon.boosts);
+				var multiply = 0;
+				for (var i = 0; i < boosts.length; i++) {
+					if (pokemon.boosts[boosts[i]] && pokemon.boosts[boosts[i]] > 0) multiply++;
+				}
+				basePower = 20 + 20 * multiply;
+			}
+			if (move.id === 'trumpcard') {
+				basePower = 40;
+				if (move.pp === 1) basePower = 200;
+				else if (move.pp === 2) basePower = 80;
+				else if (move.pp === 3) basePower = 60;
+				else if (move.pp === 4) basePower = 50;
+			}
+			if (move.id === 'venoshock') {
+				if (target.status === 'psn' || target.status === 'tox') {
+					basePower *= 2;
+					basePowerComment =' (Boosted by status)';
+				}
+			}
+			if (move.id === 'wakeupslap') {
+				if (target.status === 'slp') {
+					basePower *= 2;
+					basePowerComment = ' (Boosted by status)';
+				}
+			}
+			if (move.id === 'weatherball' && thereIsWeather) {
+				basePower = 100;
+			}
+			// Moves that check opponent speed.
+			if (move.id === 'electroball') {
+				var template = target;
+				var min = 0;
+				var max = 0;
+				if (target.volatiles.formechange) template = Tools.getTemplate(target.volatiles.formechange[2]);
+				var minRatio = (pokemon.stats.spe / this.getTemplateMinSpeed(template, target.level));
+				var maxRatio = (pokemon.stats.spe / this.getTemplateMaxSpeed(template, target.level));
+				if (minRatio >= 4) min = 150;
+				else if (minRatio >= 3) min = 120;
+				else if (minRatio >= 2) min = 80;
+				else if (minRatio >= 1) min = 60;
+				else min = 40;
+				if (maxRatio >= 4) max = 150;
+				else if (maxRatio >= 3) max = 120;
+				else if (maxRatio >= 2) max = 80;
+				else if (maxRatio >= 1) max = 60;
+				else max = 40;
+				// Special case due to being a range. Other moves are checked by technician below.
+				basePower = 0;
+				if (pokemon.ability === 'technician') {
+					if (min <= 60) min *= 1.5;
+					if (max <= 60) max *= 1.5;
+					basePowerComment = '' + ((min === max) ? max : min + ' to ' + max) + ' (Technician boosted)';
+				} else {
+					basePowerComment = (min === max) ? max : min + ' to ' + max;
+				}
+			}
+			if (move.id === 'gyroball') {
+				var template = target;
+				if (target.volatiles.formechange) template = Tools.getTemplate(target.volatiles.formechange[2]);
+				var min = (Math.floor(25 * this.getTemplateMinSpeed(template, target.level) / pokemon.stats.spe) || 1);
+				var max = (Math.floor(25 * this.getTemplateMaxSpeed(template, target.level) / pokemon.stats.spe) || 1);
+				if (min > 150) min = 150;
+				if (max > 150) max = 150;
+				// Special case due to range as well.
+				basePower = 0;
+				if (pokemon.ability === 'technician') {
+					if (min <= 60) min *= 1.5;
+					if (max <= 60) max *= 1.5;
```

Comment:
I'm not sure I understand, if the `max` calculated equals `60` it will be multiplied by 1.5 since I'm using `<=`, which leads to `90`, but the actual max for Gyro Ball, technician or not, is `150`. Am I missing something?